docs/travis.md

```diff
@@ -0,0 +1,102 @@
+# Travis CI
+
+The Zulip server uses [Travis CI](https://travis-ci.org/) for its
+continuous integration.  This page documents useful tools and tips to
+know about when using Travis CI and debugging issues with it.
+
+## Goals
+
+The overall goal of our Travis CI setup is to catch as many possible
+future bugs in Zulip as possible, minimizing delay and false positives
+that waste the time of developers around the Zulip community
+debugging.  There are a few implications of this overall goal:
+
+* If a test is failing nondeterministically in Travis CI, we consider
+  that to be an urgent problem.
+* If the tests become a lot slower, that is also an urgent problem.
+* Everything we do in CI should be to the extent possible runnable in
+  an environment where one can iterate quickly, so that developers
+  only need to wait 10 minutes for Travis CI to run repeatedly when
+  modifying the Travis CI configuration itself.
+
+## Configuration
+
+The main Travis configuration file is `.travis.yml`.  The specific
+test suites we have are listed in the `matrix` section in that file.
+We've configured it to use a few helper scripts for each job:
+
+* `tools/travis/setup-$TEST_SUITE`: A Zulip script that sets up the
```

Comment:
The `TEST_SUITE` variable needs explanation here if you're going to refer to it.

Which seems like a good idea -- probably good to explain it anyway.